### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'

### ISSUE THREAD

**Issue by mrA2Z0101** [Upsonic/Upsonic#139]
Hello, I was trying to install the GPT Computer Assistant and ran into the following error:
 × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-_i6g05ss\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-_i6g05ss\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-_i6g05ss\overlay\Lib\site-packages\setuptools\build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-_i6g05ss\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 6, in <module>
      FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**I am running windows 10 and using Python 3.12.3**

**Comment by onuratakan** [comment]
Can you retry with this command

```console
pip3 install gpt-computer-assistant[default]
```

**Comment by mrA2Z0101** [comment]
I tried it and I got the same error.
I included what it shows when installing it as well here:
C:\Users\AKZis>cd gpt-computer-assistant

C:\Users\AKZis\gpt-computer-assistant>pip3 install gpt-computer-assistant[default]
Collecting gpt-computer-assistant[default]
  Using cached gpt_computer_assistant-0.6.12.tar.gz (163 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.12 does not provide the extra 'default'
INFO: pip is looking at multiple versions of gpt-computer-assistant[default] to determine which version is compatible with other requirements. This could take a while.
  Using cached gpt_computer_assistant-0.6.11.tar.gz (154 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.11 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.10.tar.gz (153 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.10 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.9.tar.gz (152 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.9 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.8.tar.gz (148 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.8 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.7.tar.gz (144 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.7 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.6.tar.gz (139 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.6 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.5.tar.gz (136 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.5 does not provide the extra 'default'
INFO: pip is still looking at multiple versions of gpt-computer-assistant[default] to determine which version is compatible with other requirements. This could take a while.
  Using cached gpt_computer_assistant-0.6.4.tar.gz (141 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.4 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.3.tar.gz (139 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.3 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.2.tar.gz (138 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.2 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.1.tar.gz (126 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.1 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.6.0.tar.gz (120 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.6.0 does not provide the extra 'default'
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
  Using cached gpt_computer_assistant-0.5.6.tar.gz (113 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.6 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.5.tar.gz (112 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.5 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.4.tar.gz (110 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.4 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.3.tar.gz (108 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.3 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.2.tar.gz (105 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.2 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.1.tar.gz (106 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.1 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.5.0.tar.gz (103 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.5.0 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.4.0.tar.gz (99 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.4.0 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.3.0.tar.gz (91 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.3.0 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.2.0.tar.gz (11 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.2.0 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.1.2.tar.gz (10 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.1.2 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.1.1.tar.gz (10 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
WARNING: gpt-computer-assistant 0.1.1 does not provide the extra 'default'
  Using cached gpt_computer_assistant-0.1.0.tar.gz (10 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "c:\users\akzis\appdata\local\programs\python\python38-32\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-pcj_sjw7\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-pcj_sjw7\overlay\Lib\site-packages\setuptools\build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-pcj_sjw7\overlay\Lib\site-packages\setuptools\build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "C:\Users\AKZis\AppData\Local\Temp\pip-build-env-pcj_sjw7\overlay\Lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 6, in <module>
      FileNotFoundError: [Errno 2] No such file or directory: 'requirements.txt'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
